src/Model/Widget/SpanParser.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpTui\Tui\Model\Widget;
+
+use PhpTui\Tui\Model\Color;
+use PhpTui\Tui\Model\Color\AnsiColor;
+use PhpTui\Tui\Model\Color\RgbColor;
+use PhpTui\Tui\Model\Modifier;
+use PhpTui\Tui\Model\Style;
+
+final class SpanParser
```

Comment:
let's make this a subset of the Symfony Console component markup, and add a class comment here with a link to: `https://symfony.com/doc/current/console/coloring.html`